plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/PodPortValidator.java

```diff
@@ -16,73 +16,93 @@
+	private static final int PORT_MAXLENGTH = 63;
 
-	private static final Pattern POD_PORT_REGEXP = Pattern.compile("[a-z0-9]([a-z0-9-]*[a-z0-9])*");
+	private static final Pattern PORT_REGEXP = Pattern.compile("[a-z0-9]([a-z0-9-]*[a-z0-9])*");
 
-	private static final IStatus POD_PORT_FORMAT_ERROR_STATUS = ValidationStatus
+	private static final IStatus PORT_FORMAT_ERROR_STATUS = ValidationStatus
 			.error("The Pod port must be a unique integer\nor a name matching [a-z0-9]([a-z0-9-]*[a-z0-9])*");
 
 	private static final IStatus CANCEL_STATUS = ValidationStatus
 			.cancel("The Pod port must be a unique integer\nor the name of a port in the backend pods.");
 
-	private static final IStatus POD_PORT_UNIQUE_ERROR_STATUS = ValidationStatus
+	private static final IStatus PORT_UNIQUE_ERROR_STATUS = ValidationStatus
 			.error(NLS.bind(ServicePortDialog.UNIQUE_ERROR, "pod"));
-	private static final IStatus POD_PORT_LENGTH_ERROR_STATUS = ValidationStatus
-			.error("Pod port name exceeds the maximum length " + POD_PORT_MAXLENGTH);
-	private static final IStatus POD_PORT_INTERVAL_ERROR_STATUS = ValidationStatus
+	private static final IStatus PORT_LENGTH_ERROR_STATUS = ValidationStatus
+			.error("Pod port name exceeds the maximum length " + PORT_MAXLENGTH);
+	private static final IStatus PORT_INTERVAL_ERROR_STATUS = ValidationStatus
 			.error("Pod port number is out of valid interval from 0 to 65535");
 
 	private final String podPort;
+	private final List<ServicePortAdapter> ports;
 
-	private final List<IServicePort> ports;
-
-	public PodPortValidator(final String podPort, final List<IServicePort> ports) {
+	public PodPortValidator(final String podPort, final List<ServicePortAdapter> ports) {
 		this.podPort = podPort;
 		this.ports = ports;
 	}
 
 	@Override
 	public IStatus validate(final Object value) {
-		// port cannot be empty
-		if (StringUtils.isEmpty(value)) {
+		if (!(value instanceof String)
+				|| StringUtils.isEmpty(value)) {
 			return CANCEL_STATUS;
 		}
-		final String newPodPort = (String) value;
-		if (!newPodPort.equals(podPort)) {
-			try {
-				final long portNumber = Long.valueOf(value.toString());
-				// validate range
-				if (portNumber < 0 || portNumber > 65535) {
-					return POD_PORT_INTERVAL_ERROR_STATUS;
-				}
-			} catch (NumberFormatException e) {
-				// port name must match the regular expression
-				// TODO: validate against backend pod ports.
-				if (newPodPort.length() > POD_PORT_MAXLENGTH) {
-					return POD_PORT_LENGTH_ERROR_STATUS;
-				}
-				if (!POD_PORT_REGEXP.matcher(newPodPort).matches()) {
-					return POD_PORT_FORMAT_ERROR_STATUS;
-				}
+
+		IStatus status = Status.OK_STATUS;
+		final String port = (String) value;
+		if (!port.equals(podPort)) {
+			status = validateValue(port);
+			if (status.isOK()) {
+				status = validateUniqueness(port);
+			}
+		}
+		return status;
+	}
+
+	private IStatus validateUniqueness(final String port) {
+		for (IServicePort existing : ports) {
```

Comment:
can be simpler:`if ports.contains(existing.getTargetPort())`

Reply:
@bdshadow this unfortunately wont work. ports contains IServicePort objects. What we get as parameter is a string. We need to check if among the service ports objects, there's already one whose target port matches the given (String) parameter.